openedx_ledger/models.py

```diff
@@ -297,22 +297,74 @@ class Meta:
             "e.g. a LearnerCreditEnterpriseCourseEnrollment ID."
         ),
     )
-    reference_type = models.CharField(
+    subsidy_access_policy_uuid = models.UUIDField(
+        blank=True,
+        null=True,
+        help_text="A reference to the subsidy access policy which was used to create a transaction for the content."
+    )
+    history = HistoricalRecords()
+
+
+class ExternalTransactionReference(TimeStampedModel):
+    """
+    Model of references to transactions from an external system and their associated Transaction objects.
+
+    .. no_pii:
+    """
+
+    class Meta:
+        """
+        Metaclass for ExternalTransactionReference.
+        """
+
+        unique_together = [('external_reference_id', 'external_reference_type')]
+        index_together = [('external_reference_id', 'external_reference_type')]
+
+    transaction = models.ForeignKey(
+        Transaction,
+        related_name='external_reference',
+        null=True,
+        on_delete=models.CASCADE,
+        help_text=(
+            "The Transaction to which this external reference is associated."
+        ),
+    )
+    external_reference_id = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        primary_key=True,
+        db_index=True,
+        help_text=(
+            "The name of the external reference.  This is typically the name of the system that the reference is "
+            "associated with."
+        ),
+    )
+    external_reference_type = models.CharField(
```

Comment:
If we make a model called `ExternalTransactionReferenceType`, we could add new types without writing code.  So imagine that model exists and has columns `id`, `name`, and maybe `slug` (or whatever else we need).
The `external_reference_type` field you've defined here could then become a FK _to_ this other type model.  This would also help us de-couple external backends (in this case, GEAG) from our code.
Also, maybe the name of the type model is a little more generic, like `ExternalFulfillmentProvider` or something?

This type of design also helps address this concern in your ADR:
>Additionally, the current implementation does not easily allow for additional financial allocation sources without subsequent migrations.
(In the current state of your PR, we'd still need a migration every time we want to add a new value to `TransactionReferenceTypeChoices`)

Reply:
I'm not exactly sure what you mean by the name of the type model is a little more generic- are you asking that we not refer to the reference source/type as `ExternalTransactionReferenceType`?

Reply:
models and ADR have been updated